### PR TITLE
Subject Descriptions in Subject Filter

### PR DESCRIPTION
**Pull request by megandouli**
[cache.zip](https://github.com/sandboxnu/searchneu/files/5627285/cache.zip)
# what
Want the longer subject description (ex: Computer Science) in addition to the subject abbreviation (ex: CS) in the subject filter

# how
Added a subjects table to the database
Added an optional description property to relevant types
Changed the CSS styling so description shows up on a separate line

# Instructions to test this
run 
yarn db:migrate
yarn db:refresh

would recommend using unzipped cache folder above in place of the cache you have

run yarn scrape

### CODE REVIEW

**Review by epillay — APPROVED** [review]
amazing stuff on the frontend :meow-wow:

**Review by puneetjohal — APPROVED** [review]
Looks and works great! Thanks for picking up this ticket Megan :)

**Review by beccajohnson17 — APPROVED** [review]
this is a 10000x increase in the usability of the subject filter.

File: frontend/components/ResultsPage/filters.ts [hunk 1]
```diff
@@ -87,6 +87,7 @@ export type FilterOptions = Record<FilteredKeys<FilterSpecs, FilterSpec<'Dropdow
 export type Option = {
   value: string,
   count: number
```
Comment (gamburgm) [inline frontend/components/ResultsPage/filters.ts]: can you add a comma here :^)

File: backend/scrapers/classes/parsersxe/subjectAbbreviationParser.js [hunk 2]
```diff
@@ -46,8 +50,16 @@ function processSubjectListResponse(subjects) {
...
+      description: he.decode(subject.description),
+      subjectCode: subject.code,
+    };
+  });
+  subjects = _.keyBy(subjects, 'subjectCode');
+  return _.mapValues(subjects, 'description');
+}
+
+// Export for testing https://philipwalton.com/articles/how-to-unit-test-private-functions-in-javascript/
+export { createDescriptionTable as _createDescriptionTable }
```
Comment (gamburgm) [inline backend/scrapers/classes/parsersxe/subjectAbbreviationParser.js]: Can you add a semicolon here?

File: backend/searcher.ts [hunk 3]
```diff
@@ -102,15 +103,17 @@ class Searcher {
...
+        this.subjects[obj.abbreviation] = obj.description;
+      });
     }
   }
 
   /**
    * return a set of all existing subjects of classes
    */
-  getSubjects(): Set<string> {
+  getSubjects(): {} {
```
Comment (gamburgm) [inline backend/searcher.ts]: Can you change this to the same name as the type of the `subjects` property?

File: backend/searcher.ts [hunk 4]
```diff
@@ -235,7 +238,13 @@ class Searcher {
       resultCount: results[0].hits.total.value,
       took: results[0].took,
       aggregations: _.fromPairs(filters.map((filter, idx) => {
-        return [filter, results[idx + 1].aggregations[filter].buckets.map((aggVal) => { return { value: aggVal.key, count: aggVal.doc_count } })];
+        return [filter, results[idx + 1].aggregations[filter].buckets.map((aggVal) => {
```
Comment (gamburgm) [inline backend/searcher.ts:241]: Can we re-format this for clarity? e.g.
```
return [
    filter,
    results[idx + 1].aggregations[filter].buckets.map((aggVal) => {
...
];
```

Honestly, might make sense to break the aggregations generation into a helper function that we invoke in the map.

**Review by gamburgm — CHANGES_REQUESTED** [review]
We should add :chungus: to GitHub reacts :beyes:

**Review by gamburgm — APPROVED** [review]
Bet :+1:

**Pull request merged**